nosedep.py

```diff
@@ -275,21 +274,36 @@ def prepare_suite(self, suite):
     def prepareTest(self, test):
         """Prepare and determine test ordering"""
         all_tests = {}
-        for t in test:
-            for tt in t:
-                if tt.context is None:
-                    continue
-                try:
-                    if isinstance(tt, ContextSuite):  # MethodTestCase
-                        all_tests[tt.context.__name__] = self.prepare_suite(tt)
-                        setattr(all_tests[tt.context.__name__], 'nosedep_run', True)
-                    else:  # FunctionTestCase
-                        all_tests[tt.test.test.__name__] = tt
-                except AttributeError:
-                    # This exception is confusing - reraise the original one
-                    reraise(tt.test.exc_class, tt.test.exc_val, tt.test.tb)
+
+        # When passing a directory to nose we have an extra
+        # top level that we need to enter.
+        all_tests = self.find_next_level(test, all_tests)
         return self.orderTests(all_tests, test)
 
+    def find_next_level(self, test, all_tests):
+        """Find test level of ContextSuite object
+
+        find_next_level() is a re-entrant function use to find the parent directory of a test or group of tests.
```

Comment:
nit: I think you mean "recursive" rather than "re-entrant"... if I recall correctly, re-entrant means that more than one thread can be executing the piece of code at one time...